include/taskolib/GitRepository.h

```diff
@@ -0,0 +1,175 @@
+ *                    have the shape "OLD_NAME -> NEW_NAME"
+ * \param handling:   Handling status [unchanged, unstaged, staged, untracked, ignored]
+ * \param changes:    Change status [new file, deleted, renamed, typechanged, modified, unchanged, ignored, untracked]
+ */
+typedef struct
+{
+    std::string path_name;
+    std::string handling;
+    std::string changes;
+} filestatus;
+
+/**
+ * A class to wrap used methods from C-Library libgit2.
+ * 
+ * This class is necessary to handle C-pointers in a way that
+ * C++ programmers using this class do not have to think about them.
+ * 
+ * This class does not reflect the full libgit2 library.
+ * git functions which are not implemented in this class will be
+ * implemented in case of necessity
+ */
+class GitRepository
+{
+public:
+
+    /**
+     * Constructor which specifies the root dir of the git repository.
+     * \param file_path: path to "sequences"-folder (or customized directory)
+     */
+    explicit GitRepository(std::filesystem::path file_path);
+
+    /// returns member variable, which is the root dir of the git repository.
+    std::filesystem::path get_path() const;
+
+    /// returns a C-type pointer to the current loaded repository.
+    git_repository* get_repo();
+
+    /// stage all new and changed files in the repository environment.
+    void add();
+
+    /**
+     * Stage specific files listed in filepaths.
+     * \param filepaths: list of files. Either relative to repository root or absolute.
+     * \return list of indices. An index from the filepaths vector is returned if
+     *          the staging of the file failed. Returns an empty vector if all
+     *          files were staged successfully.
+     */
+   std::vector <int> add_files(std::vector<std::filesystem::path> filepaths);
+
+    /**
+     * return the commit message of ther HEAD commit.
+     * \note this function is solely for Unittest purposes
+     * \return message of last commit (=HEAD)
+     */
+    std::string get_last_commit_message();
+
+    /**
+     * Commit staged changes to the master branch of the git repository.
+     * \param commit_message: customized message for the commit
+     */
+    void commit(const std::string& commit_message);
+
+    /**
+     * Deletes seq_repository and all files within.
+     * \param seq_directory: directory holding sequence
+     * \attention seq_directory is a relative path with repo_path_ as root
+     *            if the sequence to be deleted is in "sequences/"
+     *            then set seq_directory = SEQUENCE_NAME
+     */
+    void remove_sequence(std::filesystem::path seq_directory);
+
+    /**
+     * returns current git status.
+     * \return vector of file status for each file.
+     */ 
+    std::vector<filestatus> status();
+
+    /// Destructor
+    ~GitRepository();
+
+private:
+    /// Pointer which holds all infos of the active repository.
+    LibGitPointer<git_repository*> repo_{ nullptr };
```

Comment:
With the wrapper there is no more need for the initialization to nullptr, it does that automatically.